src/main/java/com/squareup/javapoet/AnnotationSpec.java

```diff
@@ -98,6 +104,62 @@ private void emitAnnotationValues(CodeWriter codeWriter, String whitespace,
     codeWriter.emit(whitespace + "}");
   }
 
+  public static AnnotationSpec get(Annotation annotation) {
+    return get(annotation, false);
+  }
+
+  public static AnnotationSpec get(Annotation annotation, boolean includeDefaultValues) {
+    Builder builder = builder(annotation.annotationType());
+    try {
+      Method[] methods = annotation.annotationType().getDeclaredMethods();
+      Arrays.sort(methods, new MethodNameComparator());
+      for (Method method : methods) {
+        Object value = method.invoke(annotation);
+        if (!includeDefaultValues) {
+          if (Objects.deepEquals(value, method.getDefaultValue())) {
+            continue;
+          }
+        }
+        if (value.getClass().isArray()) {
+          for (int i = 0; i < Array.getLength(value); i++) {
+            addAnnotationValue(builder, method.getName(), Array.get(value, i));
+          }
+          continue;
+        }
+        if (value instanceof Annotation) {
+          AnnotationSpec nested = get((Annotation) value);
+          if (nested.members.isEmpty()) {
+            builder.addMember(method.getName(), "@$T", nested.type);
+          } else {
+            String s = nested.toString();
+            s = s.substring(s.indexOf('(') + 1, s.lastIndexOf(')'));
+            builder.addMember(method.getName(), "@$T($L)", nested.type, s);
+          }
+          continue;
+        }
+        addAnnotationValue(builder, method.getName(), value);
+      }
+    } catch (Exception e) {
+      throw new IllegalStateException("Reflecting " + annotation + " failed!", e);
+    }
+    return builder.build();
+  }
+
+  private static Builder addAnnotationValue(Builder builder, String name, Object value) {
+    if (value instanceof Class<?>) return builder.addMember(name, "$T.class", value);
+    if (value instanceof Enum) return builder.addMember(name, "$T.$L", value.getClass(), value);
+    if (value instanceof String) return builder.addMember(name, "\"$L\"", value);
```

Comment:
I think you want `$S` instead of `$L` here. Otherwise bad things happen if the string has quote characters.